### ISSUE TITLE
could I use the checkpoint in model training？

### ISSUE THREAD

**Issue by mintuos** [westlake-repl/SaprotHub#79]
Great job!
As you can see, I want to fine-tune my model based on your models, so I downloaded all the code. Fortunately, it worked.
However, I still have one issue: I don't know how to save checkpoints at each epoch, or how to load weights from the checkpoint of my fine-tuned model.
This would be helpful because our servers may break during training.
So, dear author, could you tell me if I can do this by myself?
Maybe I could modify something in the config (YAML) or somewhere else?

**Comment by mintuos** [comment]
I found that the model saves itself at each epoch, so maybe I can continue training from the saved model（Trained by yourself on ColabSaprot）.
Is that correct? I ran some tests, and the loss seems reasonable.

**Comment by LTEnjoy** [comment]
Hi!

Did you train your model based on Colab interface or you just ran your training through python scripts?

Of course you could load previous weights and continually train your model. If you train your model through Colab interface, you could load your model from the option ``Trained by yourself on ColabSaprot`` and continually train it. If you mannually run the training through the YAML config file, you could load your checkpoint by adding an argument ``from_checkpoint`` and set it to the path to your checkpoint.

**Comment by mintuos** [comment]
I copied and modified the repository and the saprothub.ipynb to run locally, similar to how it's done in Google Colab. However, the code doesn't save a .pt file, which I need in order to use the 'from_checkpoint' argument for breakpoint training. So I have one more question: can both the 'Trained by yourself on ColabSaprot' option and the 'from_checkpoint' argument achieve the same outcome for breakpoint training?

**Comment by LTEnjoy** [comment]
If you run ColabSaprot locally, it will not generate a .pt file as ColabSaprot adopts LoRA technique to fine-tune a model (.pt file is saved for full fine-tuning). However, you can still load the checkpoint by using ``Trained by yourself on ColabSaprot`` option. It will automatically load the LoRA module and enable you to keep on training.

**Comment by mintuos** [comment]
thank you!
